### DISCUSSION TITLE
Discuss!

### DISCUSSION THREAD

**Discussion by larsbrinkhoff** [PDP-10/its#1982] [General]
Hey all,

GitHub has a new feature for discussions.  This is the first post there.

**Comment by decuser** [comment]
I have found some references to logo and in exploring the system there appear to be several different logos present - llogo, bbn logo, apple ii logo?, etc. Is there a list of them somewhere and how to invoke/use them along with any documentation? Also, AI apparently had a Turtle Logo that displayed graphically, are there notes pertaining to how to get that up and running?

**Comment by larsbrinkhoff** [comment]
Logo is pretty well documented Alexey Slyusar's markdown file.  Here are some pointers:
- https://github.com/PDP-10/its/blob/master/doc/logo.md
- https://gunkies.org/wiki/Logo
- https://github.com/larsbrinkhoff/its-manual/wiki/The-Logo-programming-language
- https://github.com/pdp11/sits/wiki/List-of-Logo-memos

Several of them started with floor turtles.  Turtle graphics on displays was developed as a more cost effective way to provide turtles for children, and this became strongly associated with Logo from then on.  Arguably, this was not by design.

The various implementations differ widely in how they provide turtle graphics.  Please try the above references and come back with notes!

**Reply by decuser** [reply]
Shonuff, that's great info. I was able to get turtle graphics on the knight tv!

**Comment by DCressey** [comment]
Hello, everybody.  I'd like to join the discussion.  I'm David Cressey.  In 1971-72, I was one of five people who designed and built Muddle, the programming language on which Zork was built.  My role in Muddle was pretty mundane.  I was responsible for memory management, especially the automatic garbage collector. I still remember it as one of the more interesting programming challenges I got involved with.

I don't remember much about Muddle, but I'll help where I can.  

Meanwhile,  I am on github,  at Github.com/Dcressey.  I've got a few repos over there,  mainly expand-csv.  Expand-csv is a toy template engine, written in Powershell.

**Comment by larsbrinkhoff** [comment]
Hello David/@DCressey,

Welcome here!  There isn't much discussions here, but feel free to start one if you like.  I have invited you to the PDP-10 group, although membership isn't a requirement in anyway.

Maybe you can see something that piques your interest here?  
https://github.com/PDP-10/its/issues?q=label%3Amuddle 
https://github.com/PDP-10/its/issues?q=label%3A%22dynamic+modeling%22

**Comment by eswenson1** [comment]
Also, you and I should chat a bit (via email?). I have a set of things that need "doing" regarding Muddle.  One thing specifically, is to get the whole package system, library system, compiler, and assembler working for MDL 56. Currently, it only works for MDL 55.  RRS and I have made some progress, and we can provide details.

**Comment by eswenson1** [comment]
I've added three Muddle-related issues, which are the most pressing.  

https://github.com/PDP-10/its/issues/2241
https://github.com/PDP-10/its/issues/2239
https://github.com/PDP-10/its/issues/2240

@DCressey You may wish to talk to me about these issues via private email first so that I can catch you up on all the work that has been done so far and where the blockers are.

**Comment by DCressey** [comment]
Sorry to be so slow in responding.  I got behind in my email.  I'm not going to join a project to restore muddle at this time.  Muddle is in my past, but I don't see it in my future.  

I'm more interested in discussion than in actual work.  My brain gets easily fatigued these days.  And I realize that my personal recollections are not as valuable as other people's real research.  But that's wehre I am at.  

I do wish you guys well,  and if there's a way I can track your progress, that will be fun for me.

    On Monday, October 9, 2023 at 01:28:58 PM EDT, Eric Swenson ***@***.***> wrote:  
 
 


I've added three Muddle-related issues, which are the most pressing.

#2241
#2239
#2240

@DCressey You may wish to talk to me about these issues via private email first so that I can catch you up on all the work that has been done so far and where the blockers are.
